reset-pass/account-validator-tool/src/java/org/sakaiproject/accountvalidator/tool/controller/MainController.java

```diff
@@ -0,0 +1,1283 @@
+import org.springframework.context.MessageSource;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.ResponseBody;
+import org.json.simple.JSONObject;
+import org.sakaiproject.authz.api.SecurityAdvisor;
+import org.sakaiproject.entitybroker.EntityReference;
+import org.sakaiproject.user.api.Authentication;
+import org.sakaiproject.user.api.AuthenticationException;
+import org.sakaiproject.user.api.Evidence;
+import org.sakaiproject.user.api.UserAlreadyDefinedException;
+import org.sakaiproject.user.api.UserEdit;
+import org.sakaiproject.user.api.UserLockedException;
+import org.sakaiproject.user.api.UserPermissionException;
+import org.sakaiproject.util.ExternalTrustedEvidence;
+import org.sakaiproject.authz.api.SecurityService;
+import org.sakaiproject.entitybroker.DeveloperHelperService;
+import org.sakaiproject.user.api.AuthenticationManager;
+import org.sakaiproject.exception.IdUnusedException;
+
+import javax.servlet.http.HttpServletRequest;
+import org.sakaiproject.emailtemplateservice.service.EmailTemplateService;
+import org.sakaiproject.event.api.UsageSessionService;
+import javax.servlet.http.HttpServletResponse;
+import org.sakaiproject.accountvalidator.logic.ValidationException;
+import org.sakaiproject.accountvalidator.model.ValidationClaim;
+import org.sakaiproject.authz.api.AuthzGroupService;
+import org.sakaiproject.site.api.Site;
+import org.sakaiproject.site.api.SiteService;
+import org.sakaiproject.site.api.ToolConfiguration;
+import org.sakaiproject.util.IdPwEvidence;
+
+@Slf4j
+@Controller
+public class MainController {
+    
+    @Autowired
+    private ServerConfigurationService serverConfigurationService;
+    
+    @Autowired
+    private MessageSource messageSource;
+    
+    @Autowired
+    private SessionManager sessionManager;
+    
+    @Autowired
+    private PreferencesService preferencesService;
+    
+    @Autowired
+    private ValidationLogic validationLogic;
+    
+    @Autowired
+    private UserDirectoryService userDirectoryService;
+    
+    @Autowired
+    private SecurityService securityService;
+    
+    @Autowired
+    private DeveloperHelperService developerHelperService;
+    
+    @Autowired
+    private AuthenticationManager authenticationManager;
+    
+    @Autowired
+    private HttpServletRequest httpServletRequest;
+    
+    @Autowired
+    private UsageSessionService usageSessionService;
+    
+    @Autowired
+    private EmailTemplateService emailTemplateService;
+    
+    @Autowired
+    protected SiteService siteService;
+    
+    @Autowired
+    protected AuthzGroupService authzGroupService;
+    
+    private static final String TEMPLATE_KEY_ACKNOWLEDGE_PASSWORD_RESET = "acknowledge.passwordReset";
+    private static final String MAX_PASSWORD_RESET_MINUTES = "accountValidator.maxPasswordResetMinutes";
+    private static final int MAX_PASSWORD_RESET_MINUTES_DEFAULT = 60;
+
+    private Locale userLocale;
```

Comment:
I don't like defining this variables at the controler, define and use them at request level, no problem having the same pattern again and again.